src/requisitionMerge.js

```diff
@@ -20,6 +20,67 @@ const MERGE_FIELDS_MAPPING = {
   quantityRequested: 'Cust_stock_order',
 };
 
+/**
+ * Regimen columns which should be altered in the merging process.
+ * Keys are columns of the outgoing requisition, values are
+ * columns of the incoming requisition.
+ */
+const MERGE_REGIMENS_MAPPING = {
+  patients_adultes_recus: 'patientsOnTreatmentAdult',
+  patients_enfants_recus: 'patientsOnTreatmentChildren',
+  nouvelle_inclusion_adulte: 'patientsToInitiateTreatmentAdult',
+  nouvelle_inclusion_enfant: 'patientsToInitiateTreatmentChild',
+  regimen_value: 'value',
+  regimen_comment: 'comment',
+  referes: 'remarks',
+};
+
+/**
+ * Convert mSupply indicators data to eSIGL regimen lines.
+ * @param {Object} regimenData
+ * {
+ *   indicators: [ { "ID": ..., "code": ..., "program_ID": ... }, ... ],
+ *   rows: [ { "ID": ..., "code": ..., "indicator_ID": ... }, ... ],
+ *   columns: [ { "ID": ..., "code": ..., "indicator_ID" ... }, ... ],
+ *   values: [ { "ID": ..., "column_ID": ..., "row_ID": ..., "value": ... }, ... ],
+ * }
+ * @return {Object} [ { code: ..., columnCodeA: ..., columnCodeB: ..., ... }, ... ]
+ */
+const createRegimenLineItems = regimenData => {
+  const { indicators, rows, columns, values } = regimenData;
+  // Convert each indicator to set of associated regimen lines.
+  const regimenLineItems = indicators
+    .map(indicator => {
+      const indicatorRows = rows.filter(
+        ({ indicator_ID: rowIndicatorID }) => rowIndicatorID === indicator.ID
+      );
+      const indicatorColumns = columns.filter(
+        ({ indicator_ID: columnIndicatorID }) => columnIndicatorID === indicator.ID
+      );
+      // Convert each indicator row to matching eSIGL regimen line.
+      const indicatorRegimenLineItems = indicatorRows.map(indicatorRow => {
+        const { code } = indicatorRow;
+        const regimenColumns = indicatorColumns.map(indicatorColumn => {
+          const [{ value: columnValue }] = values.filter(
```

Comment:
why not `.find`?

Reply:
28be16180f14adbccdd47500f34994e121417a0f.